practice-two/src/pages/Analytics/Analytics.jsx

```diff
@@ -65,31 +69,14 @@ const Analytics = () => {
   };
 
   // Fetch data from the server when the component mounts
-  useEffect(() => {
-    const getAllCustomers = async () => {
-      setIsLoading(true);
-      const data = await getAllCustomerService();
-      if (data instanceof Error) {
-        setIsError(data.message);
-        toast.error("Having some error. Please try again!", {
-          position: "top-right",
-          autoClose: 5000,
-          hideProgressBar: false,
-          closeOnClick: true,
-          pauseOnHover: true,
-          draggable: true,
-          progress: undefined,
-          theme: "dark",
-        });
-      } else {
-        setCustomers(data);
-        setIsError(null);
-      }
-      setIsLoading(false);
-    };
-
-    getAllCustomers();
-  }, []);
+  const {
+    data: customers,
+    error: isError,
+    isLoading,
+  } = useSWR(`${BASE_URL}/${PATH}`, getAllCustomerService, {
+    refreshInterval: 5000,
+    shouldRetryOnError: false,
```

Comment:
why set it false?

Reply:
I used it because of avoiding errors being called back too many times, and instead, I'll handle that error separately in my code.